messenger/multiple-buses.rst

```diff
@@ -0,0 +1,41 @@
+.. index::
+    single: Messenger; Multiple buses
+
+Multiple Buses
+==============
+
+A common architecture when building application is to separate commands from
+queries. Commands are actions that do something and queries fetches data. This
+is called CQRS (Command Query Responsibility Segregation). See Martin Fowler's
+`article about CQRS`_ to learn more. This architecture could be used together
+with the messenger component by defining multiple buses.
+
+A **command bus** is a little different form a **query bus**. As example,
+command buses must not return anything and query buses are rarely asynchronous.
+You can configure these buses and their rules by using middlewares.
+
+It might also be a good idea to separate actions from reactions by introducing
+an **event bus**. The event bus could have zero or more subscribers.
```

Comment:
If we're mentioning that events have zero or more subscribers, we should probably mention that commands and queries have exactly one handler.